core/controllers/resources_test.py

```diff
@@ -528,10 +529,43 @@ def setUp(self):
             'flac': ['audio/flac']
         }
 
+        # The HANDLER_ARGS_SCHEMA dict in AudioUploadHandler uses the
+        # ACCEPTED_AUDIO_EXTENSIONS as a key value in the
+        # 'should_end_with' validator. The dict needs to be swapped
+        # seperately as it is a class variable.
+        mock_audio_upload_handler_args_schema = {
+            'POST': {
+                'raw_audio_file': {
+                    'schema': {
+                        'type': 'basestring',
+                        'validators': [{
+                            'id': 'is_nonempty'
+                        }]
+                    }
+                },
+                'filename': {
+                    'schema': {
+                        'type': 'basestring',
+                        'validators': [{
+                            'id': 'should_end_with',
+                            'choices': [
+                                '.' + extension for extension in list(
```

Comment:
Hi, @JayVivarekar can you use a variable, instead of the entire logic inside the schema. 
Also, I guess the error that you are facing is because of this choices part (as I have pulled your branch and removed the choices and everything seems to be working perfectly.)

So, one thing I want from your side -- can you check by logging the choices whether they are the same as expected.

Thank you.

Reply:
@Nik-09 I logged the choices as you suggested and they are as expected. Attaching the screenshot below.
![Capture5](https://user-images.githubusercontent.com/10340981/165781705-79e1d8a8-d1da-49ed-b213-4d2622ba37c2.PNG)

However, I was not able to reproduce the result which you mentioned (removing the choices). I am still getting a 302 error. Can you share your logs too?